src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.Integer.cs

```diff
@@ -0,0 +1,261 @@
+        public static void ReadInt64_OutOfRangeValues_ShouldThrowOverflowException(string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            Assert.Throws<OverflowException>(() =>
+            {
+                var reader = new CborValueReader(data);
+                reader.ReadInt64();
+            });
+        }
+
+        [Theory]
+        [InlineData("20")] // -1
+        [InlineData("3863")] // -100
+        [InlineData("3b7fffffffffffffff")] // long.MinValue
+        public static void ReadUInt64_OutOfRangeValues_ShouldThrowOverflowException(string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            Assert.Throws<OverflowException>(() =>
+            {
+                var reader = new CborValueReader(data);
+                reader.ReadUInt64();
+            });
+        }
+
+        [Theory]
+        [InlineData("40")] // empty text string
+        [InlineData("60")] // empty byte string
+        [InlineData("f6")] // null
+        [InlineData("80")] // []
+        [InlineData("a0")] // {}
+        [InlineData("f97e00")] // NaN
+        [InlineData("fb3ff199999999999a")] // 1.1
+        public static void ReadInt64_InvalidTypes_ShouldThrowInvalidOperationException(string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            InvalidOperationException exn = Assert.Throws<InvalidOperationException>(() =>
+            {
+                var reader = new CborValueReader(data);
+                reader.ReadInt64();
+            });
+
+            Assert.Equal("Data item major type mismatch.", exn.Message);
+        }
+
+        [Theory]
+        [InlineData("40")] // empty byte string
+        [InlineData("60")] // empty text string
+        [InlineData("f6")] // null
+        [InlineData("80")] // []
+        [InlineData("a0")] // {}
+        [InlineData("f97e00")] // NaN
+        [InlineData("fb3ff199999999999a")] // 1.1
+        public static void ReadUInt64_InvalidTypes_ShouldThrowInvalidOperationException(string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            InvalidOperationException exn = Assert.Throws<InvalidOperationException>(() =>
+            {
+                var reader = new CborValueReader(data);
+                reader.ReadUInt64();
+            });
+
+            Assert.Equal("Data item major type mismatch.", exn.Message);
+        }
+
+        [Theory]
+        [InlineData("00")] // 0
+        [InlineData("17")] // 23
+        [InlineData("40")] // empty byte string
+        [InlineData("60")] // empty text string
+        [InlineData("f6")] // null
+        [InlineData("80")] // []
+        [InlineData("a0")] // {}
+        [InlineData("f97e00")] // NaN
+        [InlineData("fb3ff199999999999a")] // 1.1
+        public static void ReadCborNegativeIntegerEncoding_InvalidTypes_ShouldThrowInvalidOperationException(string hexEncoding)
```

Comment:
Also needs the 3c/3d/3e invalid encoding tests.

Reply:
That's already covered in a different test: `ReadInt64_InvalidData_ShouldThrowFormatException`

Reply:
Ah, different method this one.